cmd/secrets/secret_helper.go

```diff
@@ -14,37 +14,66 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
-	"path/filepath"
 	"strings"
 
 	"github.com/spf13/cobra"
 
+	"github.com/DataDog/datadog-agent/cmd/secrets/providers"
 	s "github.com/DataDog/datadog-agent/pkg/secrets"
 )
 
+// This executable provides a "read" command to fetch secrets. It can be used in
+// 2 different ways:
+//
+// 1) With the "--with-provider-prefixes" option enabled. Each input secret
+// should follow this format: "providerPrefix/some/path". The provider prefix
+// indicates where to fetch the secrets from. At the moment, we support "file"
+// and "k8s_secret". The path can mean different things depending on the
+// provider. In "file" it's a file system path. In "k8s_secret", it follows this
+// format: "namespace/name/key".
+//
+// 2) Without the "--with-provider-prefixes" option. The program expects a root
+// path in the arguments and input secrets are just paths relative to the root
+// one. So for example, if the secret is "my_secret" and the root path is
+// "/some/path", the fetched value of the secret will be the contents of
+// "/some/path/my_secret". This option was offered before introducing
+// "--with-provider-prefixes" and is kept to avoid breaking compatibility.
+
 const (
-	maxSecretFileSize = 8192
+	providerPrefixesFlag    = "with-provider-prefixes"
+	providerPrefixSeparator = "/"
```

Comment:
can use a separator that is not used usually in a path like `@` or `$`?

Reply:
This is required if we want to implement what you suggested in https://github.com/DataDog/datadog-agent/pull/9132#discussion_r710835638